azurerm/resource_arm_front_door.go

```diff
@@ -172,24 +172,25 @@ func resourceArmFrontDoor() *schema.Resource {
 										Required:     true,
 										ValidateFunc: afd.ValidateBackendPoolRoutingRuleName,
 									},
+									// Remove default value for #4461
 									"cache_use_dynamic_compression": {
 										Type:     schema.TypeBool,
 										Optional: true,
-										Default:  false,
 									},
+									// Remove default value for #4461
 									"cache_query_parameter_strip_directive": {
 										Type:     schema.TypeString,
 										Optional: true,
 										ValidateFunc: validation.StringInSlice([]string{
 											string(frontdoor.StripAll),
 											string(frontdoor.StripNone),
 										}, false),
-										Default: string(frontdoor.StripNone),
```

Comment:
This is a breaking change? we can't do this until 2.0

Reply:
No, this is not a breaking change because the value is never evaluated unless the `cache_use_dynamic_compression` is set to TRUE. So this is basically a NoOp… WDYT?

Reply:
Looks like setting it to StripNone or StripAll won't change the 'caching' field but only the 'Query string caching behaviour'. Is there a way to change the 'caching' field?